tools/cache/time-defined-cache.go

```diff
@@ -92,6 +89,8 @@ func (t *TimeDefinedCache[T]) Set(value T) {
 
 func (t *TimeDefinedCache[T]) setWithoutLock(value T) {
 	t.savedValue = value
-	t.savedValueSet = true
-	t.lastRefresh = time.Now()
+
+	if t.lastRefresh == nil || t.minRefreshDuration.Nanoseconds() != 0 {
+		t.lastRefresh = k6tpointer.P(time.Now())
+	}
```

Comment:
I have to disagree on this one :)
That's exactly why the `pointer` package was introduced, to avoid having to introduce a variable just for the sake of having a pointer.